docs/articles/devops/vrille/05.comment.build.md

```diff
@@ -0,0 +1,156 @@
+---
+title: "Chapitre 2 : Le build"
+summary: "Pourquoi arrive-t-on aux limites du DevOps ? Comme pour tout, le temps fait son œuvre. La démarche DevOps n'y échappe pas. Vous pouvez avoir une démarche très aboutie, au bout de plusieurs mois ou plusieurs années, celle-ci aura naturellement du plomb dans l'aile."
+authors:
+    - Philippe Morisseau
+date: 2021-12-06
+---
+
+Maintenant que nous avons conceptualisé le principe de machine à état dans un "meta" pipeline **CI/CD**, comment faire pour que le **build** de nos différents composants puisse être géré de façon harmonieuse.
+
+Comment faire pour s'assurer qu'une évolution sur une librairie va entrainer systématiquement la recompilation de composants utilisant cette librairie ?
+
+Comment faire pour ne recompiler que ce qu'il y a besoin de recompiler ?
+
+## La conceptualisation du processus de build
+
+Pour commencer, il nous faut un référentiel de **build**. 
+
+### Le référentiel de build
+
+Ce référentiel aura pour unique but d'identifier chaque composant de notre application, avec :
+
+- le nom du **repository git**
+- la dernière version,
+- l'id du commit git lié à la dernière version,
+- les dépendances du composant
+
+!!! sample "exemple"
+    Imaginons 2 librairies : **Lib1** et **Lib2**. **Lib2** dépend de **Lib1**.
+
+    Voici ce que pourrait donner notre référentiel au format **yaml** :
+
+    ```yaml
+    components:
+    - name: lib1
+      version: 1.0
+      commit: 0123456789ab
+    - name: lib2
+      version: 1.0
+      commit: 123456789abc
+      dependencies: 
+      - lib1
+    ```
+
+Ce référentiel va évoluer au fur et à mesure des itérations pour correspondre exactement à l'état souhaité de tous les composants de l'application. Et pour permettre une historisation des évolutions au fur et à mesure des itérations, quoi de mieux qu'un **repository Git**.
+On va donc mettre notre référentiel de **build** dans un **repo.** que l'on va nommer "meta".
+
+### Le processus de build
+
+Imaginons que j'apporte une modification sur mon référentiel de **build**, cela implique que j'ai ajouté, modifié ou supprimé un ou plusieurs composants de mon application. Pour ne pas prendre de risque je vais donc parcourir chaque composant déclaré dans mon référentiel et pour chacun, vérifier :
+
+- s'il a déjà été **buildé** : il n'y aura donc rien à faire,
+- s'il y a une mise à jour : il faudra déclencher le **build** du composant,
+- s'il y a eu une mise à jour d'une des dépendances : il faudra redéclencher le **build** du composant,
+- s'il y a eu un échec d'une dépendance : il ne faudra pas lancer le **build**.
+
+C'est comme un arbre où chaque composant correspond à un noeud.
+
+!!! sample "exemple"
+    Imaginons que durant notre **sprint** nous ayons fait évoluer nos composants B, D, G et H et l'arbre de dépendance ci-dessous :
+    ![tree](../../../img/05.comment.build.002.svg)
```

Comment:
Selon moi, ce schéma manquerait d'indications quant au sens des dépendances. Si cela paraît clair au niveau le plus haut, je me perds un peu lorsque l'on descend plus bas. Surtout en ce qui concerne le composant H en échec, car j'aurais eu tendance à me dire que celui-ci s'appuie sur la compilation du composant K. Or visiblement c'est l'inverse.

Ou peut-être que je ne l'interprète pas correctement.

Reply:
J'ai ajouté des flèches et ajouter une petite explication :
"B and C use A. So, B and C depend on A."